### PR TITLE
Backward compatibility for torch 1.x and fix for windows

### PR DESCRIPTION
**Pull request by jkulhanek**
Adresses #1961

### CODE REVIEW

**Comment by machenmusik** [comment]
Thanks again @jkulhanek.
Given the PR title, I added the pyproject.toml changes to allow torch 1.13.1 as well.
If the intent was to force PyTorch 2.0 on Windows (which I am still testing) then my apologies, feel free to revert.

**Comment by machenmusik** [comment]
So far, torch 2.0.1 on windows does not train.

```
WARNING: Using a slow implementation of HashEncoding.
🏃 🏃 Install tcnn for speedups 🏃 🏃
pip install git+https://github.com/NVlabs/tiny-cuda-nn/#subdirectory=bindings/torch
```
(which was done already)

and later

```
NameError: name 'tcnn' is not defined
```

So torch 1.13.1 on Windows does appear necessary for the moment.

**Comment by machenmusik** [comment]
(Updated README as well.)

**Comment by jkulhanek** [comment]
What is the error when you import tinycudann manually?

**Comment by jkulhanek** [comment]
Disclaimer: I don’t feel strongly about this PR. I just opened it as a possibility.

**Comment by jkulhanek** [comment]
> So far, torch 2.0.1 on windows does not train.
> 
> ```
> WARNING: Using a slow implementation of HashEncoding.
> 🏃 🏃 Install tcnn for speedups 🏃 🏃
> pip install git+https://github.com/NVlabs/tiny-cuda-nn/#subdirectory=bindings/torch
> ```
> 
> (which was done already)
> 
> and later
> 
> ```
> NameError: name 'tcnn' is not defined
> ```
> 
> So torch 1.13.1 on Windows does appear necessary for the moment.

Which cuda do you have installed? What error message are you getting if you try importing tcnn manually? Is there an issue with your version of cuda and windows being tracked in the cuda repo?

File: README.md [hunk 1]
```diff
@@ -99,6 +99,14 @@ python -m pip install --upgrade pip
 
 Install pytorch with CUDA (this repo has been tested with CUDA 11.7 and CUDA 11.8) and [tiny-cuda-nn](https://github.com/NVlabs/tiny-cuda-nn)
 
+NOTE: On Windows, PyTorch 2.0 is not currently supported; please install PyTorch 1.13.1 with CUDA 11.7:
```
Comment (jkulhanek) [inline README.md:102]: @machenmusik are you sure?

Comment (machenmusik) [reply README.md:102]: I meant with Nerfstudio, mainly because I didn't get it to work with 2.0.1 when I tried it, not that it can't work

**Comment by machenmusik** [comment]
> Which cuda do you have installed? What error message are you getting if you try importing tcnn manually? Is there an issue with your version of cuda and windows being tracked in the tcnn repo?

Should be both CUDA 11.7 and 11.8.
Have not had a chance to delve further.

**Comment by jkulhanek** [comment]
@machenmusik , can you please undo the change to readme for now (until we find the cause of the problem with your setup)?

**Comment by machenmusik** [comment]
Sure - had to do manually via mobile.

Does it work on Windows with torch 2.0.1 and either cu117 or 118 for others?

**Comment by machenmusik** [comment]
> Disclaimer: I don’t feel strongly about this PR. I just opened it as a possibility.

Understood, thanks for quickly putting together a PR to try!  I have limited time to try this weekend, but will see if I can get torch 2.0.1 to work with either cu117 or 118, which would allow reverting the pyproject.toml change as well.

Bigger picture - if following https://ubuntu.com/tutorials/enabling-gpu-acceleration-on-ubuntu-on-wsl2-with-the-nvidia-cuda-platform first then the typical nerfstudio installation docs can enable nerfstudio to run on Windows machines using WSL2 Linux with expected performance, we may want to change instructions to recommend that.  But I am not sure that works either.

**Comment by jkulhanek** [comment]
Thanks! Did you previously used cuda 11.3 or 11.7 with torch 1.13 on windows?

**Comment by machenmusik** [comment]
Originally 11.3 but now 11.7

(I don't think there is a cu118 version of torch 1.13.1)

**Comment by machenmusik** [comment]
torch 2.0.1+cu117 has exact same behavior as 118
https://github.com/nerfstudio-project/nerfstudio/pull/1962#issuecomment-1555950488

So at the moment it seems torch1.13.1+cu117 is necessary on Windows.

**Review by machenmusik — APPROVED** [review]
From this morning's diagnosis
- torch 2.0.1+cu118 appears OK on Windows
  - however, torch.compile is not currently supported on Windows, so we do need these backward compatibility changes
- tinycudann may require adjustments for certain GPUs to work properly 
  - specifically, CUDA architecture 89 such as RTX4090 needs to set environment variable TCNN_CUDA_ARCHITECTURES to include 89 prior to bindings install

We may want to add a documentation note about TCNN_CUDA_ARCHITECTURES for completeness.   See https://github.com/NVlabs/tiny-cuda-nn/issues/314

**Review by tancik — APPROVED** [review]
LGTM

**Pull request merged**